plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,464 @@
+  ## specify servers via a url matching:
+  ##  [protocol://][:password]@address[:port]
+  ##  e.g.
+  ##    tcp://localhost:26379
+  ##    tcp://:password@192.168.99.100
+  ##    unix:///var/run/redis-sentinel.sock
+  ##
+  ## If no servers are specified, then localhost is used as the host.
+  ## If no port is specified, 26379 is used
+  servers = ["tcp://localhost:26379"]
+
+  ## specify server password
+  # password = "s#cr@t%"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = true
+`
+}
+
+func (r *RedisSentinel) Description() string {
+	return "Read metrics from one or many redis-sentinel servers"
+}
+
+func (r *RedisSentinel) Init() error {
+	if len(r.Servers) == 0 {
+		r.Servers = []string{"tcp://localhost:26379"}
+	}
+
+	r.clients = make([]*RedisSentinelClient, len(r.Servers))
+
+	tlsConfig, err := r.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	for i, serv := range r.Servers {
+		u, err := url.Parse(serv)
+		if err != nil {
+			return fmt.Errorf("unable to parse to address %q: %v", serv, err)
+		}
+
+		if u.Scheme != "tcp" && u.Scheme != "unix" {
+			return fmt.Errorf("invalid scheme %q. expected tcp or unix", u.Scheme)
+		}
+
+		address := u.Host
+		if u.Scheme == "unix" {
+			address = u.Path
+		}
+
+		sentinel := redis.NewSentinelClient(
+			&redis.Options{
+				Addr:      address,
+				Password:  r.Password,
+				Network:   u.Scheme,
+				PoolSize:  1,
+				TLSConfig: tlsConfig,
+			},
+		)
+
+		tags := map[string]string{}
+		if u.Scheme == "unix" {
+			tags["socket"] = u.Path
+		} else {
+			tags["source"] = u.Hostname()
+			tags["port"] = u.Port()
+		}
+
+		r.clients[i] = &RedisSentinelClient{
+			sentinel: sentinel,
+			tags:     tags,
+		}
+	}
+
+	return nil
+}
+
+// Redis list format has string key/values adjacent, so convert to a map for easier use
+func toMap(vals []interface{}) map[string]string {
+	m := make(map[string]string)
+
+	for idx := 0; idx < len(vals)-1; idx += 2 {
+		key, keyOk := vals[idx].(string)
+		value, valueOk := vals[idx+1].(string)
+
+		if keyOk && valueOk {
+			m[key] = value
+		}
+	}
+
+	return m
+}
+
+func castFieldValue(value string, fieldType configFieldType) (interface{}, error) {
+	var castedValue interface{}
+	var err error
+
+	switch fieldType {
+	case configFieldTypeFloat:
+		castedValue, err = strconv.ParseFloat(value, 64)
+	case configFieldTypeInteger:
+		castedValue, err = strconv.ParseInt(value, 10, 64)
+	case configFieldTypeString:
+		castedValue = value
+	default:
+		return nil, fmt.Errorf("unsupported field type %v", fieldType)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed cating value %v: %v", value, err)
+	}
+
+	return castedValue, nil
+}
+
+func prepareFieldValues(
+	fields map[string]string,
+	configFieldTypeMap map[string]configFieldType,
+) (map[string]interface{}, error) {
```

Comment:
If the params fit in 120 cols we should not wrap
```suggestion
func prepareFieldValues(fields map[string]string, typeMap map[string]configFieldType) (map[string]interface{}, error) {
```